core/admin.py

```diff
@@ -83,28 +88,60 @@ class AllocationSourceAdmin(admin.ModelAdmin):
         "compute_allowed",
     )
 
+    def save_model(self, request, obj, form, change):
+        request.data = {"renewal_strategy": obj.renewal_strategy,
+                        "name": obj.name,
+                        "compute_allowed": obj.compute_allowed}
+        if not change:
+            api = AllocationSourceViewSet()
+            api.create(request)
+        else:
+            request.data = {}
+
+            # renewal strategy modified
+            if form.initial['renewal_strategy'] != obj.renewal_strategy:
+                request.data['renewal_strategy'] = obj.renewal_strategy
+
+            # compute allowed modified
+            if form.initial['compute_allowed'] != obj.compute_allowed:
+                request.data['compute_allowed'] = obj.compute_allowed
+
+            if request.data:
+                api = AllocationSourceViewSet()
+                api.update(request, obj.uuid)
+
+            # if allocation source is end dated
+            if form.initial['end_date'] != obj.end_date:
+                api = AllocationSourceViewSet()
+                api.perform_destroy(obj, request=request)
+
+    def has_delete_permission(self, request, obj=None):
+        return False
```

Comment:
Yea I agree, we have never used API views/serializers from within the backend ( or have we ? ) so I wasn't sure how to go about it. Using requests would have been more cumbersome, but an abstraction layer which deals with this sort of situation would be perfect in the future